tests/unit/htdocs/class/IframeBreakoutTest.php

```
@@ -0,0 +1,52 @@
+    #[Test]
+    public function sourcePatternExcludesQuotesAndAngleBracketsFromSrc(): void
+    {
+        $src = (string) file_get_contents(XOOPS_ROOT_PATH . '/class/textsanitizer/iframe/iframe.php');
+        // The src group (3rd capture) must use the [^"'<>] character class.
+        self::assertStringContainsString('([^\\"\'<>]*)', $src, 'iframe src capture is not hardened');
+    }
+
+    #[Test]
+    public function benignIframeStillMatches(): void
+    {
+        $ok = preg_match(self::PATTERN, '[iframe=300]https://example.com/embed[/iframe]', $m);
```

PHPMD (2.15.0)

[warning] Avoid variables with short names like $ok. Configured minimum length is 3. (undefined)

(ShortVariable)